### PR TITLE
Migrate designer to normalized props

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio-designer/issues/696

Here added normalized props logic to transactions and replaced useAllUserProps with computed store.

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
- [ ] hi @rpominov, I need you to do
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/shared/nano-states/nano-states.ts
Comment (rpominov) [inline apps/designer/app/shared/nano-states/nano-states.ts:78]: Does `propsIndexStore` get recomputed only when `propsStore` change? 

Looking at this code it seems like we might be creating separate instances of the computed store per `instanceId`.

Did you test how this works in practice?

File: apps/designer/app/canvas/canvas.tsx
Comment (rpominov) [inline apps/designer/app/canvas/canvas.tsx:27]: (writing here just to make it a thread)

Bug:

- add a "button" to canvas
- set "type" props to "reset"
- select the button on canvas
- copy (Cmd+C)
- click on canvas to make "body" selected
- paste (Cmd+V)
- the "type" props value is not copied

Same bug when selecting in the navigator instead of canvas

File: apps/designer/app/canvas/shared/instance.ts
Comment (TrySound) [reply apps/designer/app/canvas/shared/instance.ts:67]: Yes, though its data used on server side for now.

File: apps/designer/app/shared/nano-states/nano-states.ts
Comment (TrySound) [reply apps/designer/app/shared/nano-states/nano-states.ts:78]: Yes, propsIndexStore is recomputed every time propsStore is changed. Each instancePropsStore is recomputed when propsIndexStore is changed. Though useStore triggers rerender only when one of props array elements are changed.

Ideally we need to maintain a separate state with index though it's complicated with patches so we recompute shared index for all invocations instead.

File: apps/designer/app/canvas/canvas.tsx
Comment (TrySound) [reply apps/designer/app/canvas/canvas.tsx:27]: Yeah, we need to generate new id for each inserted prop.

File: apps/designer/app/shared/sync/sync-stores.ts
Comment (kof) [inline apps/designer/app/shared/sync/sync-stores.ts:48]: I would just call the old props `deprecatedProps`

**Review by kof — APPROVED** [review]

File: apps/designer/app/canvas/canvas.tsx
Comment (kof) [reply apps/designer/app/canvas/canvas.tsx:27]: I approved, though if its a new bug please fix it before merging

Comment (TrySound) [reply apps/designer/app/canvas/canvas.tsx:27]: Kinda new. Right now a lot of duplicated prop.id are stored in db. Here the bug just become visible.

Comment (TrySound) [reply apps/designer/app/canvas/canvas.tsx:27]: I'll fix props and styles copy paste in next PR

**Pull request merged**
